odoo/addons/base/models/qweb.py

```diff
@@ -939,7 +939,10 @@ def _compile_directive_content(self, el, options, indent):
         body = []
         if el.getchildren():
             for item in el:
-                if not isinstance(item, etree._Comment):
+                if isinstance(item, etree._Comment):
+                    if options.get('preserve_comments'):
+                        self._appendText(etree.tostring(etree.Comment(item.text)).decode(), options)
```

Comment:
That seems way more complicated than necessary. Why not just reserialise `item` directly? Or even serialise it by hand, XML comments are actually more restrictive than HTML ones, so if the document parsed properly you can just dump the content of the comment inbetween comment markers.